custodial-copy-backend/src/main/scala/uk/gov/nationalarchives/custodialcopy/OcflService.scala

```diff
@@ -92,6 +91,16 @@ class OcflService(ocflRepository: OcflRepository, semaphore: Semaphore[IO]) {
       MissingAndChangedObjects(missingObjects, changedObjects)
     }
 
+  private def isChecksumUnchanged(fixitiesMap: java.util.Map[DigestAlgorithm, String], checksums: List[Checksum]): Boolean = {
+    if (fixitiesMap.size() != checksums.size) {
+      false
+    } else {
+      checksums.forall(eachChecksum =>
+        Some(fixitiesMap.get(DigestAlgorithm.fromOcflName(eachChecksum.algorithm.toLowerCase))).contains(eachChecksum.fingerprint)
```

Comment:
On line 64, If you change `ocflFileObject.getFixity` to `ocflFileObject.getFixity.asScala.toMap`, then you can remove the `Some` here and make it a bit cleaner since Scala's `.get` returns an `Option` Also, if you use Scala 3's syntax, you can remove quite a few braces

```
  private def isChecksumUnchanged(fixitiesMap: Map[DigestAlgorithm, String], checksums: List[Checksum]): Boolean =
    if (fixitiesMap.size != checksums.size) then false
    else checksums.forall(
      eachChecksum => fixitiesMap.get(DigestAlgorithm.fromOcflName(eachChecksum.algorithm.toLowerCase)).contains(eachChecksum.fingerprint)
    )
```

Reply:
I had simply converted it from a singular to plural and extracted a function out for that check. While we are at it, do you know why it compares as "contains" rather than "equals"?

Reply:
Because `Option` is a container type. When you do `.get` on a Scala Map, you get an Option; if you did `fixitiesMap(DigestAlgorithm.fromOcflName(eachChecksum.algorithm.toLowerCase))` (without the `.get`), you could do `equals` (or `==`) but risk getting a `java.util.NoSuchElementException` if the key is not there.

Reply:
I understand about the `Option`,, that's not a problem. 
I think you misunderstood my question, which was unrelated to the changes but related to the code on those lines. I was wondering why the existing code was doing a `contains` comparison rather than `equals` comparison?